utils/python/CIME/XML/entry_id.py

```diff
@@ -56,6 +56,40 @@ def get_type_info(self, vid):
             return None
         else:
             return self._get_type_info(node)
+    
+    def get_default(self, node):
+        default = self.get_optional_node("default_value", root=node)
+        if default is not None:
+            return default.text
+        else:
+            return None
+        
+    
+    # Get type description , expect child with tag "type" for node         
+    def get_type (self, node):
+        type_node = self.get_optional_node("type", root=node)
+        if type_node is not None:
+            return type_node.text
+        else:
+            # Default to string
+            return "char"
+            
+    # Get description , expect child with tag "description" for parent node
+    def get_description (self, node):
+        type_node = self.get_optional_node("desc", root=node)
+        if type_node is not None:
+            return type_node.text
+        else:
+            return None
+
+    # Get group , expect child with tag "group" for parent node        
+    def get_group (self, node):
```

Comment:
The idea is that subroutines with an initial underscore are only called interior to this module or other modules within the XML directory.  node should never be referenced outside the XML directory so I think that you are correct, this should be _get_group